extension.js

```diff
@@ -466,26 +514,71 @@ const ClipboardIndicator = Lang.Class({
         let that = this;
 
         Clipboard.get_text(CLIPBOARD_TYPE, function (clipBoard, text) {
-            that._processClipboardContent(text);
+            if (text === null && SAVE_IMAGE) {
+                Clipboard.get_content(CLIPBOARD_TYPE, SUPPORT_PNG ? 'image/png' : 'image/jpeg', function (clipboard, content) {
+                    that._processClipboardContent(content.get_data(), 'image', content.hash());
+                })
+            }
+            else {
+                const itens = text.split('\n');
+                let blockText = false;
+                let hasText = false;
+
+                itens.map((values) => {
+                    // Dolphin debugger
+                    if (!values.length) return;
+
+                    // ebp === webp
+                    // vif === avif
+                    // peg === jpeg
+                    const support_image_type = ['png', 'jpg', 'peg', 'gif', 'ico', 'ebp', 'svg', 'vif'];
+                    const type = values.substring(values.length - 3);
+
+                    if (support_image_type.indexOf(type) !== -1) {
+                        // Dolphin return file:// + $PATH.
+                        const path = values.substring(0, 7) === 'file://' ? values.substring(7) : values;
+                        
+                        const file = Gio.file_new_for_path(path);
+                        if (!file.query_exists(null)) {
+                            hasText = true;
+                        }
+                        else {
+                            const bytes = file.load_bytes(null)[0]
+                            that._processClipboardContent(bytes, 'image', bytes.hash());
+                        }
+                    }
+                    else {
+                        hasText = true;
+                    }
+
+                    if (hasText && !blockText) {                            
+                        blockText = true
+                        that._processClipboardContent(text, "text");
+                    }
+                })
+            }
         });
     },
 
-    _processClipboardContent (text) {
+    // type most be image or text
+    _processClipboardContent (byteContent, type, name) {
         const that = this;
-
-        if (STRIP_TEXT) {
-            text = text.trim();
+        let content;
+        if (STRIP_TEXT && type === "text") {
```

Comment:
I think we're missing the case where STRIP_TEXT is false here. Shouldn't `content` be set to `byteContent` regardless?

Reply:
e.g. ```        if (type === "text") {
            if (!STRIP_TEXT) {
                content = byteContent;
            }
            else {
                content = byteContent.trim();
            }
        }
        else {
            content = writerImage(byteContent, name);
        }
```